MiniAn/minian_main.py

```diff
@@ -467,6 +472,110 @@ def load_doric_to_xarray(
+    file_ref.close()
+    
+    # Apply shifts to spatial footprints of each session
+    A_shifted = apply_transform(A_concat.chunk(dict(height = -1, width = -1)), shiftds["shifts"])
+
+    window = shiftds['temps_shifted'].isnull().sum('session')
+    window, _ = xr.broadcast(window, shiftds['temps_shifted'])
+    def set_window(wnd):
+        return wnd == wnd.min()
+    window = xr.apply_ufunc(set_window, window, input_core_dims=[["height", "width"]], 
+                            output_core_dims=[["height", "width"]], vectorize=True)
+
+    # Calculate centroids of spatial footprints for cells inside a window.
+    cents = calculate_centroids(A_shifted, window)
+
+    # Calculate pairwise distance between cells in all pairs of sessions. 
+    # Note that at this stage, since we are computing something along the session dimension, 
+    # it is no longer considered as a metadata dimension, so we remove it    
+    dist = calculate_centroid_distance(cents, "session", [])
+
+    # Threshold centroid distances, keeping only cell pairs with distance less than param_dist.
+    param_dist = minian_parameters.params_cross_reg["param_dist"]
+
+    dist_ft = dist[dist["variable", "distance"] < param_dist].copy()
+    dist_ft = group_by_session(dist_ft)
+
+    # Generate mappings for ids of the current and reference sessions
+    mappings = calculate_mapping(dist_ft)
+    mappings_meta = resolve_mapping(mappings)
+    mappings_meta_fill = fill_mapping(mappings_meta, cents)
+
+    # Update unit ids of the current spatial componenets A
+    ids        = list(A["unit_id"].values)
+    new_ids    = [0]*len(ids)
+    ref_id_max = int(A_ref.coords["unit_id"].values.max()) + 1
+
+    for i in range(len(mappings_meta_fill)):
+        # Matching ids between the sessions
+        if mappings_meta_fill.iloc[i]["group"][0] == ("current", "reference"):
+            index = ids.index(mappings_meta_fill.iloc[i]["session"]["current"])
+            new_ids[index] = int(mappings_meta_fill.iloc[i]["session"]["reference"])
+        # Unique ids for the current session
+        elif mappings_meta_fill.iloc[i]["group"][0] == ("current",):
+            index = ids.index(mappings_meta_fill.iloc[i]["session"]["current"])
+            new_ids[index] = ref_id_max
+            ref_id_max += 1
+
+    A["unit_id"] = new_ids
+
+    return A
+
+
+def get_footprints(filename, rois_h5path, dims):
+    
+    with h5py.File(filename, 'r') as file_:
+
+        roi_names =  list(file_.get(rois_h5path))
+
+        roi_ids = np.zeros((len(roi_names) - 1))
+        footprints = np.zeros(((len(roi_names) - 1), dims["height"].size, dims["width"].size), np.float64)
+        for i in range(len(roi_names) - 1):
+            attrs = utils.load_attributes(file_, f"{rois_h5path}/{roi_names[i]}")
+
+            roi_ids[i] = int(attrs["ID"])
+
+            coords = np.array(attrs["Coordinates"])
+            mask = np.zeros((dims["height"].size, dims["width"].size), np.float64)
```

Comment:
Is it working without the conversion to BGR?

Reply:
yes all looks fine to me. mapping is correct.